### PR TITLE
workflow: Update README

### PR DESCRIPTION
**Pull request by andrewwormald**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,144 +12,361 @@
...
-- Consumer management and graceful shutdown
-- Supports event streaming platforms such as Kafka and Reflex (or you can write your own implementation of the EventStreamer interface!)
-- Built in support for timeout operations (e.g. account cool down periods etc).
-- Built in support for callbacks (e.g. Call an async endpoint and trigger the callback from a webhook handler).
-- Connect two workflows together. Wait for specific events from another workflow and make it part of your workflow!
-- Super Duper testable
-
+# Workflow
+ Workflow is an event driven workflow that allows for robust, durable, and scalable sequential business logic to
+ be executed in a deterministic manner.
```
Comment (echarrod) [inline README.md:17]: Looks like this is being rendered as a heading, is that intentional?

Comment (andrewwormald) [reply README.md:17]: Hm strange it wasnt on my machine 🤔

File: README.md [hunk 2]
```diff
@@ -12,144 +12,361 @@
...
+- **Event fusion:** Add event connectors to your workflow to consume external event streams (even if its from a different event streaming platform).  
+- **Callbacks:** Allow for manual callbacks from webhooks or manual triggers from consoles to progress the workflow such as approval buttons or third-party webhooks.  
+- **Parallel consumers:** Specify how many step consumers should run or specify the default for all consumers. 
+- **Consumer management:** Consumer management and graceful shutdown of all processes making sure there is no goroutine leaks!
+
+---
+
+## Installation
+To start using workflow you will need to add the workflow module to your project. You can do this by running:
+```go
```
Comment (echarrod) [inline README.md:35]: Think these should be shell, as they'll be run from terminal, and aren't Go code (colouring doesn't look right as a result)
image


```suggestion
```shell
```

File: README.md [hunk 3]
```diff
@@ -12,144 +12,361 @@
...
+defer cancel()
+
+record, err := wf.Await(ctx, foreignID, runID, StepThree)
+if err != nil {
+	...
+}
+```
+
+### Detailed examples
+Head on over to `./example` to get familiar with **callbacks**, **timeouts**, **testing**, **connectors** and
```
Comment (echarrod) [inline README.md:166]: Can make this a link so it's clickable? 
```suggestion
Head on over to [./example](./example)] to get familiar with **callbacks**, **timeouts**, **testing**, **connectors** and
```

File: README.md (hunk 2 above)
Comment (andrewwormald) [reply README.md:35]: oh silly me. Good catch

File: README.md [hunk 4]
```diff
@@ -12,144 +12,361 @@
...
+	direction LR
+    
+	[*]-->Initiated
+
+    Initiated-->Running
+    
+    Running-->Completed
+    Running-->Paused
+
+    Paused-->Running
```
Comment (echarrod) [inline README.md:189]: Can we label the run states here? I wasn't sure what it was, had to look in the mermaid code 👍

File: README.md (hunk 1 above)
Comment (andrewwormald) [reply README.md:17]: This is the rendering I get

Screenshot 2024-08-13 at 16 05 08

File: README.md [hunk 5]
```diff
@@ -12,144 +12,361 @@
...
+    ...,
+    StepTwo,
+).WithOptions(
+    workflow.PauseAfterErrCount(3),
+)
+```
+---
 ## Example
 
 Head on over to `./examples` to get familiar with the syntax 😊
```
Comment (echarrod) [inline README.md:334]: Same would apply here I guess!

File: README.md [hunk 6]
```diff
@@ -12,144 +12,361 @@
...
-
-30. **Topic:**
-    - A method that generates a topic for producing events in the event streamer based on the workflow name and status.
-
-31. **TimerFunc:**
-    - A function type used to calculate the time for a timeout. It takes a context, record, and the current time as inputs.
-
-32. **Trigger:**
-    - A method in the workflow API that initiates a workflow for a specified foreignID and starting status. It returns a runID and allows for additional configuration options.
+**Builder:**
```
Comment (echarrod) [inline README.md:338]: This is quite a lot to take in! Would adding it to a table make it a little more digestible, and easier to lookup the key?

File: README.md (hunk 1 above)
Comment (echarrod) [reply README.md:17]: I'm looking in the browser at https://github.com/luno/workflow/tree/andreww-updateReadMe, is that what you're looking at?

Comment (echarrod) [reply README.md:17]: I think removing the space before, or maybe also adding a line in between the two, should fix

```suggestion
Workflow is an event driven workflow that allows for robust, durable, and scalable sequential business logic to
be executed in a deterministic manner.
```

**Pull request merged**